ngraph_bridge/ngraph_utils.cc

```diff
@@ -463,6 +475,198 @@ bool IsProcessedByNgraphPass(Graph* g) {
+
+  if (str_path.back() != '/') {
+    str_path += "/";
+  }
+
+  MakeTBDir(str_path);
+
+  std::string dname = GetSessionName(file_idx, nodes);
+  str_path += (dname.insert(0, "ngraph") + "/");
+
+  MakeTBDir(str_path);
+
+  str_path += ("ngraph" + to_string(file_idx));
+
+  TF_RETURN_IF_ERROR(CreateSummaryFromGraphDef(
+      graph_def, str_path));  // create tensorflow event
+
+  return Status::OK();
+}
+
+Status UpdateComputeTime(int file_idx, std::string cluster,
+                         std::string sess_name, int step, int compute_time) {
+  const char* path = std::getenv("NGRAPH_TF_TB_LOGDIR");
+
+  if (path == nullptr) {
+    return Status::OK();
+  }
+
+  std::string str_path(path);
+  if (str_path.back() != '/') {
+    str_path += "/";
+  }
+
+  std::string dname(sess_name);
+  str_path += (dname.insert(0, "stats") + "/");
+
+  MakeTBDir(str_path);
+
+  struct dirent* dp;
+  vector<std::string> files;
+
+  // inspect directory's files
+  try {
+    DIR* dir = opendir(str_path.c_str());
+
+    while ((dp = readdir(dir)) != nullptr) {
+      if (dp->d_type == DT_REG)  // look at all files in directory
+      {
+        files.push_back(std::string(dp->d_name));
+      }
+    }
+
+    closedir(dir);
+  } catch (...) {
+    return errors::Internal("Error inspecting files in directory: " + str_path);
+  }
+  // done inspecting directory's files
+
+  // create event object
+  tensorflow::Event event;
+  tensorflow::Summary::Value* summary_value;
+  tensorflow::Env* env = Env::Default();
+
+  event.set_wall_time(env->NowMicros() / 1e6);
+  event.set_step(step);
+  summary_value = event.mutable_summary()->add_value();
+  summary_value->set_tag(dname + " Compute Time (ms)/" + cluster);
+  summary_value->set_simple_value((float)compute_time);
+  // done creating event object
+
+  if (files.empty())  // create new tf event file
+  {
+    str_path += ("stats" + to_string(file_idx));
+
+    tensorflow::EventsWriter writer(str_path);
+
+    try {
+      writer.WriteEvent(event);
+    } catch (...) {
+      return errors::Internal("Error writing event to new event file");
+    }
+  } else  // append to tf event file in dir
+  {
+    std::string file_path = str_path + files[0];
+
+    // open tf event file at location file_path
+    std::unique_ptr<tensorflow::WritableFile> writable_file;
+
+    env->NewAppendableFile(file_path, &writable_file);
+    tensorflow::io::RecordWriter record_writer(writable_file.get());
+    // done opening tf event file at location file_path
+
+    try {
+      record_writer.WriteRecord(event.SerializeAsString());
+    } catch (...) {
+      return errors::Internal("Error updating event file with new event");
+    }
+  }
+
+  return Status::OK();
+}
+
+Status CreateSummaryFromGraph(tensorflow::Graph* graph,
+                              std::string filename_prefix) {
+  // convert Graph* to GraphDef*
+  tensorflow::GraphDef* gdef = nullptr;
+  graph->ToGraphDef(gdef);
+
+  // create event summary writer
```

Comment:
Since u convert to graph def, lets use `CreateSummaryFromGraphDef` inside `CreateSummaryFromGraph` to avoid code duplication

Reply:
Also is the `CreateSummaryFromGraph` function used anywhere? If not we can remove it completely